xbmc/video/dialogs/GUIDialogAudioSelect.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *  Copyright (C) 2005-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUIDialogAudioSelect.h"
+
+#include "Application.h"
+#include "GUIUserMessages.h"
+#include "LangInfo.h"
+#include "ServiceBroker.h"
+#include "URL.h"
+#include "Util.h"
+#include "addons/AddonManager.h"
+#include "cores/IPlayer.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "filesystem/AddonsDirectory.h"
+#include "filesystem/Directory.h"
+#include "filesystem/File.h"
+#include "filesystem/SpecialProtocol.h"
+#include "filesystem/StackDirectory.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIKeyboardFactory.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "input/Key.h"
+#include "settings/Settings.h"
+#include "settings/lib/Setting.h"
+#include "utils/JobManager.h"
+#include "utils/LangCodeExpander.h"
+#include "utils/StringUtils.h"
+#include "utils/URIUtils.h"
+#include "utils/Variant.h"
+#include "utils/log.h"
+#include "video/VideoDatabase.h"
+
+#include "system.h"
+
+using namespace ADDON;
+using namespace XFILE;
+
+#define STRING_AUDIO_HEADER 460
+#define STRING_TOAST_TEXT_ERROR 416
+
+CGUIDialogAudioSelect::CGUIDialogAudioSelect(void) : CGUIDialog(WINDOW_DIALOG_AUDIO_SELECT, "")
+{
+  m_loadType = KEEP_IN_MEMORY;
+}
+
+CGUIDialogAudioSelect::~CGUIDialogAudioSelect(void)
+{
+}
+
+bool CGUIDialogAudioSelect::OnMessage(CGUIMessage& message)
+{
+  return CGUIDialog::OnMessage(message);
+}
+
+void CGUIDialogAudioSelect::OnInitWindow()
+{
+  ShowAudioSelect();
```

Comment:
Are you sure it is correct to show the dialog in `OnInitWindow` - while the dialog is not yet fully initialized? Did you copy this pattern from an existing dialog implementation?